### PR TITLE
Clang Module XCTest support on macOS

### PR DESCRIPTION
**Pull request by masters3d**
This addresses 
https://bugs.swift.org/browse/SR-3413
https://bugs.swift.org/browse/SR-3408

This only works on macOS with the Objective-C XCTest. 

Tested with this. 
https://github.com/masters3d/SwiftPackageManagerSamples/tree/master/ObjC-PackageAcronym

```
Compile AcronymTests AcronymTest.m
Linking AcronymPackageTests
Test Suite 'All tests' started at 2016-12-14 09:57:15.866
Test Suite 'AcronymPackageTests.xctest' started at 2016-12-14 09:57:15.867
Test Suite 'AcronymTest' started at 2016-12-14 09:57:15.867
Test Case '-[AcronymTest testAcronymAbbreviateTest1]' started.
Test Case '-[AcronymTest testAcronymAbbreviateTest1]' passed (0.000 seconds).
Test Case '-[AcronymTest testAcronymAbbreviateTest2]' started.
Test Case '-[AcronymTest testAcronymAbbreviateTest2]' passed (0.000 seconds).
Test Case '-[AcronymTest testAcronymAbbreviateTest3]' started.
Test Case '-[AcronymTest testAcronymAbbreviateTest3]' passed (0.000 seconds).
Test Case '-[AcronymTest testAcronymAbbreviateTest4]' started.
Test Case '-[AcronymTest testAcronymAbbreviateTest4]' passed (0.000 seconds).
Test Case '-[AcronymTest testAcronymAbbreviateTest5]' started.
Test Case '-[AcronymTest testAcronymAbbreviateTest5]' passed (0.000 seconds).
Test Case '-[AcronymTest testAcronymAbbreviateTest6]' started.
Test Case '-[AcronymTest testAcronymAbbreviateTest6]' passed (0.000 seconds).
Test Suite 'AcronymTest' passed at 2016-12-14 09:57:15.868.
	 Executed 6 tests, with 0 failures (0 unexpected) in 0.001 (0.001) seconds
Test Suite 'AcronymPackageTests.xctest' passed at 2016-12-14 09:57:15.868.
	 Executed 6 tests, with 0 failures (0 unexpected) in 0.001 (0.001) seconds
Test Suite 'All tests' passed at 2016-12-14 09:57:15.868.
	 Executed 6 tests, with 0 failures (0 unexpected) in 0.001 (0.002) seconds
Program ended with exit code: 0
```

### CODE REVIEW

File: Sources/Build/Command.link(ClangModule).swift [hunk 1]
```diff
@@ -44,7 +44,15 @@ extension Command {
...
             args += ["-shared"]
-        case .Test, .Library(.Static):
+        case .Test:
+            #if os(macOS)
+                args += ["-Xlinker", "-bundle"]
+                args += ["-F", try platformFrameworksPath().asString]
+            #else
+                fatalError("Can't build \(product.name), \(product.type) is not yet supported.")
+            #endif
+
```
Comment (aciidb0mb3r) [inline Sources/Build/Command.link(ClangModule).swift]: Can you add a macOS only functional test case for this

**Comment by abertelrud** [comment]
Looks good to me, but I agree with @aciidb0mb3r that this should have unit tests.

File: Tests/FunctionalTests/ClangModuleTests.swift [hunk 2]
```diff
@@ -137,6 +137,34 @@ class ClangModulesTestCase: XCTestCase {
...
     }
+    
+    func testObjectiveCPackageWithTestTarget(){
+#if os(macOS)
+        fixture(name: "ClangModules/ObjCmacOSPackage") { prefix in
+            // Build the package.
+            XCTAssertBuilds(prefix)
+            XCTAssertFileExists(prefix.appending(components: ".build", "debug", "ObjCmacOSPackage".soname))
+            // Run swift-test on package.
+            XCTAssertSwiftTest(prefix)
```
Comment (aciidb0mb3r) [inline Tests/FunctionalTests/ClangModuleTests.swift]: We only need the test till this point, you can remove the XCTestHelper test (we already have test for that iirc)

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

**Pull request merged**